hall-of-fame-service/src/main/java/hcmus/alumni/halloffame/repository/HallOfFameRepository.java

```diff
@@ -0,0 +1,66 @@
+import hcmus.alumni.halloffame.model.HallOfFameModel;
+
+
+public interface HallOfFameRepository extends JpaRepository<HallOfFameModel, String> {
+	Optional<IHallOfFameDto> findHallOfFameById(String id);
+	
+	Optional<HallOfFameModel> findById(String id);
+	
+	@Query("SELECT n FROM HallOfFameModel n JOIN n.status s WHERE s.id != 4 AND n.title like %:title%")
+	Page<IHallOfFameDto> searchHof(String title, Pageable pageable);
+	
+	@Query("SELECT n FROM HallOfFameModel n JOIN n.status s WHERE s.id = :statusId AND n.title like %:title%")
+	Page<IHallOfFameDto> searchHofByStatus(String title, Integer statusId, Pageable pageable);
+	
+	@Query("SELECT n FROM HallOfFameModel n WHERE n.faculty.id = :facultyId AND n.title LIKE %:title%")
+    Page<IHallOfFameDto> searchHofByFaculty(@Param("title") String title, @Param("facultyId") Integer facultyId, Pageable pageable);
+
+	@Query("SELECT n FROM HallOfFameModel n WHERE n.beginningYear = :beginningYear AND n.title LIKE %:title%")
+    Page<IHallOfFameDto> searchHofByBeginningYear(@Param("title") String title, @Param("beginningYear") Integer beginningYear, Pageable pageable);
+	
+	@Query("SELECT n FROM HallOfFameModel n JOIN n.status s JOIN n.faculty f WHERE s.id = :statusId AND f.id = :facultyId AND n.title LIKE %:title%")
+	Page<IHallOfFameDto> searchHofByStatusAndFaculty(@Param("title") String title, @Param("statusId") Integer statusId, @Param("facultyId") Integer facultyId, Pageable pageable);
+
+	@Query("SELECT n FROM HallOfFameModel n JOIN n.status s JOIN n.faculty f WHERE s.id = :statusId AND f.id = :facultyId AND n.beginningYear = :beginningYear AND n.title LIKE %:title%")
+	Page<IHallOfFameDto> searchHofByStatusAndFacultyAndBeginningYear(@Param("title") String title, @Param("statusId") Integer statusId, @Param("facultyId") Integer facultyId, @Param("beginningYear") Integer beginningYear, Pageable pageable);
+
+	@Query("SELECT n FROM HallOfFameModel n JOIN n.faculty f WHERE f.id = :facultyId AND n.beginningYear = :beginningYear AND n.title LIKE %:title%")
+	Page<IHallOfFameDto> searchHofByFacultyAndBeginningYear(@Param("title") String title, @Param("facultyId") Integer facultyId, @Param("beginningYear") Integer beginningYear, Pageable pageable);
+
+	@Query("SELECT n FROM HallOfFameModel n JOIN n.status s JOIN n.faculty f WHERE s.id = :statusId AND f.id = :facultyId AND n.beginningYear = :beginningYear AND n.title LIKE %:title%")
+	Page<IHallOfFameDto> searchHofByAll(@Param("title") String title, @Param("statusId") Integer statusId, @Param("facultyId") Integer facultyId, @Param("beginningYear") Integer beginningYear, Pageable pageable);
+
+	@Query("SELECT n FROM HallOfFameModel n JOIN n.status s WHERE s.id = :statusId AND n.beginningYear = :beginningYear AND n.title LIKE %:title%")
+	Page<IHallOfFameDto> searchHofByStatusAndBeginningYear(@Param("title") String title, @Param("statusId") Integer statusId, @Param("beginningYear") Integer beginningYear, Pageable pageable);
```

Comment:
8 cái này gộp lại dc mà. có 1 cái cmt t có gửi link tham khảo á

Reply:
phần search bên cn, có gì tham khảo nha
`	@Query("SELECT e " +
        "FROM EventModel e " +
        "LEFT JOIN e.status s " +
        "LEFT JOIN e.faculty f " +
        "WHERE (:statusId IS NULL OR s.id = :statusId) " +
        "AND (:facultyId IS NULL OR f.id = :facultyId) " +
        "AND e.title LIKE %:title%")
	Page<IEventDto> searchEvents(@Param("title") String title, @Param("statusId") Integer statusId, @Param("facultyId") Integer facultyId, Pageable pageable);`